File: Components/ASP.NET/VB/DevAV/UserControls/Charts/ProductsOpportunitiesChart.ascx.vb
Imports System
Imports DevExpress.XtraCharts
Imports DevExpress.XtraCharts.Web

Partial Public Class UserControls_Charts_ProductsOpportunitiesChart
    Inherits ChartUserControl

    Public Overrides ReadOnly Property Chart() As WebChartControl
        Get
            Return OpportunitiesChart
        End Get
    End Property

    Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs)
        OpportunitiesChart.Series(0).DataSource = DataProvider.GetOpportunitiesChartData()
    End Sub

    Protected Sub OpportunitiesChart_CustomDrawSeriesPoint(ByVal sender As Object, ByVal e As CustomDrawSeriesPointEventArgs)
        OnCustomDrawSeriesPoint(e)
    End Sub
    Protected Sub OpportunitiesChart_CustomCallback(ByVal sender As Object, ByVal e As CustomCallbackEventArgs)
        ChangeChartWidth(Convert.ToInt32(e.Parameter))
    End Sub
End Class

File: Components/ASP.NET/CS/DevAV/UserControls/Charts/ProductsOpportunitiesChart.ascx.cs
using System;
using DevExpress.XtraCharts;
using DevExpress.XtraCharts.Web;

public partial class UserControls_Charts_ProductsOpportunitiesChart : ChartUserControl {
    public override WebChartControl Chart { get { return OpportunitiesChart; } }

    protected void Page_Load(object sender, EventArgs e) {
        OpportunitiesChart.Series[0].DataSource = DataProvider.GetOpportunitiesChartData();
    }

    protected void OpportunitiesChart_CustomDrawSeriesPoint(object sender, CustomDrawSeriesPointEventArgs e) {
        OnCustomDrawSeriesPoint(e);
    }
    protected void OpportunitiesChart_CustomCallback(object sender, CustomCallbackEventArgs e) {
        ChangeChartWidth(Convert.ToInt32(e.Parameter));
    }
}
